### DISCUSSION TITLE
Is flutter_data a Good Fit?

### DISCUSSION THREAD

**Discussion by SubutaDan** [flutterdata/flutter_data#271] [Q&A] [answered]
I have read discussion #261 and I think the answer there applies to my question as well, but I would appreciate your thoughts.

I am working on a Flutter mobile app. in which users will create items which will be persisted to the cloud and retrieved from the cloud using an an HTTP API  (which I have already written).  I would like users to be able to create, edit and delete items whether the device is on-line or off-line.  Conceptually (and perhaps as the actual implementation?) I picture a queue of pending API calls corresponding to the creates/edits, where the queue would typically be empty or almost empty because the device would usually be on-line, but where the queue would grow when the device was off-line and then drain in sequence when the device came back on-line.  Is flutter_data a good choice for building something like this?

Thanks for your help.

**Comment by frank06** [comment, marked as answer]
Yes I think it is a good choice for offline-first apps, most primitives are here

**Reply by SubutaDan** [reply]
Thank you, @frank06 .
